kuma/javascript/src/header/search.jsx

```diff
@@ -4,45 +4,75 @@ import { useEffect, useRef } from 'react';
 import SearchIcon from '../icons/search.svg';
+import CloseIcon from '../icons/close.svg';
 
 type Props = {
     initialQuery: string,
 };
 
 export default function Search({ initialQuery }: Props) {
     const locale = getLocale();
+    const [showForm, setShowForm] = React.useState(false);
+    const [query, setQuery] = React.useState(initialQuery);
 
     // After our first render, set the input field's initial value
+    // and show search form
     const inputfield = useRef(null);
     useEffect(() => {
-        if (inputfield.current && initialQuery) {
-            inputfield.current.value = initialQuery;
+        if (inputfield.current && query) {
+            inputfield.current.value = query;
+            setShowForm(true);
         }
-    }, [initialQuery]);
+    }, [query, showForm]);
+
+    const handleClick = (event) => {
+        event.preventDefault();
+
+        if (showForm === true) {
+            setShowForm(false);
+            setQuery(null);
+            if (inputfield.current) {
+                inputfield.current.value = '';
+            }
+        } else {
+            setShowForm(true);
+            inputfield.current && inputfield.current.focus();
+        }
+    };
 
     return (
-        <form
-            className="header-search"
-            id="nav-main-search"
-            action={`/${locale}/search`}
-            method="get"
-            role="search"
-        >
-            <SearchIcon className="search-icon" />
-
-            <label htmlFor="main-q" className="visually-hidden">
-                {gettext('Search MDN')}
-            </label>
-            <input
-                className="search-input-field"
-                ref={inputfield}
-                type="search"
-                id="main-q"
-                name="q"
-                placeholder={gettext('Search MDN')}
-                pattern="(.|\s)*\S(.|\s)*"
-                required
-            />
-        </form>
+        <div className={`header-search ${showForm ? 'show-form' : ''}`}>
+            <form
+                id="nav-main-search"
+                action={`/${locale}/search`}
+                method="get"
+                role="search"
+            >
+                <SearchIcon className="search-icon" />
+
+                <label htmlFor="main-q" className="visually-hidden">
+                    {gettext('Search MDN')}
+                </label>
+                <input
+                    className="search-input-field"
+                    ref={inputfield}
+                    type="search"
+                    id="main-q"
+                    name="q"
+                    placeholder={gettext('Search MDN')}
+                    pattern="(.|\s)*\S(.|\s)*"
+                    required
+                />
+            </form>
+            <button className="toggle-form" onClick={handleClick}>
+                <CloseIcon className="close-icon" />
+                <SearchIcon className="search-icon" />
```

Comment:
Not sure if this might be expensive DOM wise but, could we dynamically switch between the icons as you are currently doing with the text?

Reply:
It seems like they both need to be on the page in order to animate, otherwise it just swaps out the icons. I will experiment with this some more.

Reply:
I couldn't get this to work with keeping the animation and dynamically switching icons, unless we wanted to introduce a library like React Transition Group. Let me know if you have suggestions on how to handle this.

Reply:
Minor premature optimization :) I am fine with keeping it this way. Thanks for looking into it @mindy